core/CMakeLists.txt

```
@@ -64,10 +64,14 @@ function (add_gen_events_deps target)
 endfunction (add_gen_events_deps)
 
 if (AARCH64)
-  # Required for opcode.h and codec headers, which are auto-generated on AArch64.
+  # Required for opcode_api.h and codec headers, which are auto-generated on AArch64.
   include(../make/CMake_aarch64_gen_codec.cmake)
   add_custom_target(gen_aarch64_codec DEPENDS "${AARCH64_CODEC_GEN_SRCS}")
   include_directories(BEFORE ${PROJECT_BINARY_DIR})
+  add_custom_command(TARGET gen_aarch64_codec POST_BUILD
+    COMMAND ${CMAKE_COMMAND}
+    ARGS -E copy ${PROJECT_BINARY_DIR}/opcode_api.h
```

```suggestion
    ARGS -E copy_if_different ${PROJECT_BINARY_DIR}/opcode_api.h
```
(Or something to the same effect.)

@derekbruening 
On master this header is copied every time (with updated timestamps) and results in spurious rebuilds for all code including it.

---

I see.  Submit a PR since you have it in front of you?

---

https://github.com/DynamoRIO/dynamorio/pull/4443#discussion_r549822280 :(

---

OK.  I will do it.  I recall CMake best practices wanting a target for every output.  Looks like `copy_if_different` still runs the command (which seems weird since it's POST_BUILD and the target it's supposed to be after was *not* run).  I think making a custom target and this custom command *not* be POST_BUILD might work better since this POST_BUILD doesn't seem to do what you would think.

---

#4837

---

(Note the behavior I saw was with ninja...make might be different.)

---

Thanks!
#4837 is confirmed to remove the spurious rebuilds.